owncloudApp/src/main/java/com/owncloud/android/sharees/presentation/OCShareeViewModel.kt

```diff
@@ -0,0 +1,48 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.sharees.presentation
+
+import android.accounts.Account
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import com.owncloud.android.lib.common.OwnCloudAccount
+import com.owncloud.android.lib.common.OwnCloudClientManagerFactory
+import com.owncloud.android.sharees.data.ShareeRepository
+import com.owncloud.android.sharees.data.datasources.OCRemoteShareesDataSource
+import com.owncloud.android.sharees.domain.OCShareeRepository
+import com.owncloud.android.vo.Resource
+import org.json.JSONObject
+
+class OCShareeViewModel(
+    context: Context,
+    account: Account,
+    val shareeRepository: ShareeRepository = OCShareeRepository(
```

Comment:
Why don't you declare the repository as a dependency without default value?

Reply:
Because I can pass a mocked repository whenever is needed and write something like the code below when I just want the default repository.

```
val ocShareeViewModel = OCShareeViewModel(
    context,
    account
)
```

Anyway, I was also creating the repository from https://github.com/owncloud/android/pull/2584/files#diff-6f4c3a04e328bb66190a958efea59976R133 and that's not correct, deleting it...

Reply:
I agree to pass the repository as dependency (in the constructor), but you don't convince me with the default value...

Reply:
Anyway, I will approve this PR as I know you are iterating with the arch. I  can help you to think a little bit more after holidays 😛